comp/core/tagger/impl-remote/remote.go

```diff
@@ -287,65 +286,51 @@ func (t *remoteTagger) GenerateContainerIDFromOriginInfo(originInfo origindetect
 }
 
 // queryContainerIDFromOriginInfo calls the local tagger to get the container ID from the Origin Info.
-func (t *remoteTagger) queryContainerIDFromOriginInfo(originInfo origindetection.OriginInfo) (containerID string, err error) {
-	expBackoff := backoff.NewExponentialBackOff()
-	expBackoff.InitialInterval = 50 * time.Millisecond
-	expBackoff.MaxInterval = 200 * time.Millisecond
-	expBackoff.MaxElapsedTime = 500 * time.Millisecond
-
-	err = backoff.Retry(func() error {
-		select {
-		case <-t.ctx.Done():
-			return &backoff.PermanentError{Err: errTaggerFailedGenerateContainerIDFromOriginInfo}
-		default:
-		}
-
-		// Fetch the auth token
-		if t.token == "" {
-			var authError error
-			t.token, authError = t.options.TokenFetcher()
-			if authError != nil {
-				_ = t.log.Errorf("unable to fetch auth token, will possibly retry: %s", authError)
-				return authError
-			}
+func (t *remoteTagger) queryContainerIDFromOriginInfo(originInfo origindetection.OriginInfo) (string, error) {
+	// Fetch the auth token
+	if t.token == "" {
+		var authError error
+		t.token, authError = t.options.TokenFetcher()
+		if authError != nil {
+			_ = t.log.Errorf("unable to fetch auth token, will possibly retry: %s", authError)
+			return "", authError
 		}
+	}
 
-		// Create the context with the auth token
-		queryCtx, queryCancel := context.WithCancel(
-			metadata.NewOutgoingContext(t.ctx, metadata.MD{
-				"authorization": []string{fmt.Sprintf("Bearer %s", t.token)},
-			}),
-		)
-		defer queryCancel()
-
-		// Call the gRPC method to get the container ID from the OriginInfo.
-		containerIDResponse, err := t.client.TaggerGenerateContainerIDFromOriginInfo(queryCtx, &pb.GenerateContainerIDFromOriginInfoRequest{
-			LocalData: &pb.GenerateContainerIDFromOriginInfoRequest_LocalData{
-				ProcessID:   &originInfo.LocalData.ProcessID,
-				ContainerID: &originInfo.LocalData.ContainerID,
-				Inode:       &originInfo.LocalData.Inode,
-				PodUID:      &originInfo.LocalData.PodUID,
-			},
-			ExternalData: &pb.GenerateContainerIDFromOriginInfoRequest_ExternalData{
-				Init:          &originInfo.ExternalData.Init,
-				ContainerName: &originInfo.ExternalData.ContainerName,
-				PodUID:        &originInfo.ExternalData.PodUID,
-			},
-		})
-		if err != nil {
-			t.log.Debugf("unable to generate container ID from origin info, will retry: %s", err)
-			return err
-		}
+	// Create the context with the auth token
+	queryCtx, queryCancel := context.WithCancel(
+		metadata.NewOutgoingContext(t.ctx, metadata.MD{
+			"authorization": []string{fmt.Sprintf("Bearer %s", t.token)},
+		}),
+	)
+	defer queryCancel()
+
+	// Call the gRPC method to get the container ID from the OriginInfo.
+	containerIDResponse, err := t.client.TaggerGenerateContainerIDFromOriginInfo(queryCtx, &pb.GenerateContainerIDFromOriginInfoRequest{
```

Comment:
do you know how the timeout is set?

Reply:
Yes I believe that it is done here: https://github.com/DataDog/datadog-agent/blob/main/comp/api/api/apiimpl/server_cmd.go#L127-L132

With a default timeout of 30 seconds.

Reply:
I've added it in https://github.com/DataDog/datadog-agent/pull/34021/commits/9a6a868379cd6d56d10a712c63458e46fa183992

In case of timeout it looks like:
```
2025-02-14 14:13:15 UTC | TRACE | DEBUG | (comp/core/tagger/impl-remote/remote.go:285 in func2) | unable to generate container ID from origin info: rpc error: code = DeadlineExceeded desc = context deadline exceeded
2025-02-14 14:13:15 UTC | TRACE | DEBUG | (pkg/trace/api/container_linux.go:180 in GetContainerID) | Could not generate container ID from OriginInfo: {LocalData:{ProcessID:0 ContainerID: Inode:0 PodUID:} ExternalData:{Init:false ContainerName:apm-external-data-only PodUID:51fab96d-6ad9-4fff-98ee-9ef98693304b} Cardinality: ProductOrigin:2}, err: rpc error: code = DeadlineExceeded desc = context deadline exceeded
```